src/fsharp/NicePrint.fs

```diff
@@ -1643,317 +1372,356 @@ module private TastDefinitionPrinting =
 
     let private layoutEventInfo denv amap m (e: EventInfo) =
         let staticL = if e.IsStatic then WordL.keywordStatic else emptyL
-        let nameL = wordL (tagEvent (adjustILName e.EventName))
+
+        let eventTag =
+            let tag =
+                e.EventName
+                |> adjustILName
+                |> tagEvent
+
+            match e.ArbitraryValRef with
+            | Some vref ->
+                tag |> mkNav vref.DefinitionRange
+            | None ->
+                tag
+
+        let nameL = eventTag |> wordL
         let typL = layoutType denv (e.GetDelegateType(amap, m))
         staticL ^^ WordL.keywordEvent ^^ nameL ^^ WordL.colon ^^ typL
        
     let private layoutPropInfo denv amap m (p: PropInfo) =
-        let staticL = if p.IsStatic then WordL.keywordStatic else emptyL
-        let nameL = wordL (tagProperty (adjustILName p.PropertyName))
+        match p.ArbitraryValRef with
+        | Some v ->
+            PrintTastMemberOrVals.prettyLayoutOfValOrMemberNoInst denv v.Deref
+        | None ->
+            let modifierAndMember =
+                if p.IsStatic then
+                    WordL.keywordStatic ^^ WordL.keywordMember
+                else
+                    WordL.keywordMember
+        
+            let propTag =
+                let tag =
+                    p.PropertyName
+                    |> adjustILName
+                    |> tagProperty
+
+                match p.ArbitraryValRef with
+                | Some vref ->
+                    tag |> mkNav vref.DefinitionRange
+                | None ->
+                    tag
+
+            let nameL = propTag |> wordL
             
-        let typL = layoutType denv (p.GetPropertyType(amap, m)) // shouldn't happen
-                
-        let specGetSetL =
-            match p.HasGetter, p.HasSetter with
-            | false, false | true, false -> emptyL
-            | false, true -> WordL.keywordWith ^^ WordL.keywordSet
-            | true, true -> WordL.keywordWith ^^ WordL.keywordGet^^ SepL.comma ^^ WordL.keywordSet
-
-        staticL ^^ WordL.keywordMember ^^ nameL ^^ WordL.colon ^^ typL ^^ specGetSetL
-
-    /// Another re-implementation of type printing, this time based off provided info objects.
-    let layoutProvidedTycon (denv: DisplayEnv) (infoReader: InfoReader) ad m start lhsL ty =
-      let g = denv.g
-      let tcref = tcrefOfAppTy g ty
-
-      if isEnumTy g ty then 
```

Comment:
Are enum's now just taken care of by displaying fields?

Reply:
Yep!

F#-defined:

![image](https://user-images.githubusercontent.com/6309070/87365425-06688480-c52b-11ea-82ae-95e88322e473.png)

.NET-defined:

![image](https://user-images.githubusercontent.com/6309070/87365439-14b6a080-c52b-11ea-8f9a-772fc9417b72.png)